content/blog/three-new-tidymodels-packages/index.Rmd

```diff
@@ -0,0 +1,192 @@
+    maximize(cor_spearman, low = 0.25, high = 1),
+    maximize(imp_rf_oblique, scale = 2)
+  ) |> 
+  arrange(desc(.d_overall)) |> 
+  select(-starts_with(".d_max_"))
+```
+
+Using the `scale = 2` options puts more weight on the random forest results. 
+
+Now that we've looked at filtro, next up is the important package (yes, this is what we named it). 
+
+## important
+
+The important package does two things. First, it provides yet another tool for calculating random forest-like permutation importance scores. We highly value other packages that perform these same calculations (such as [DALEX](https://modeloriented.github.io/DALEX/) and [vip](https://github.com/koalaverse/vip/)). Our rationale for creating another package for this is that we've developed interfaces for censored regression, including dynamic metrics such as Brier scores or ROC curves that evaluate models at a specific time point. These dynamic methods aren't available in other packages, and the peculiarities of these metrics make them difficult to incorporate into existing frameworks. 
+
+Other niceties about importance scores are that any metric from the yardstick methods can be used, and we have optimized parallel processing for the underlying computations. For the latter feature, we support the future and mirai packages for parallel processing. 
+
+important also has three recipe steps for supervised feature selection (similar to what Steven Pawley did with his [colino package](https://stevenpawley.github.io/colino/)). The steps are:
+
+- [`step_predictors_best()`](https://important.tidymodels.org/reference/step_predictor_best.html)
+- [`step_predictors_retain()`](https://important.tidymodels.org/reference/step_predictor_retain.html)
+- [`step_predictors_desirability()`](https://important.tidymodels.org/reference/step_predictor_desirability.html)
+
+Let's look at the last one, which mirrors our analysis above. 
+
+```{r}
+#| label: recipe
+library(important)
+goals <-
+  desirability(
+    maximize(cor_spearman, low = 0.25, high = 1),
+    maximize(imp_rf_oblique, scale = 2)
+  )
+
+yield_rec <-
+  recipe(yield ~ ., data = yield_train) |>
+  step_impute_knn(all_predictors(), neighbors = 10) |>
+  step_predictor_desirability(
+    all_predictors(),
+    score = goals,
+    prop_terms = 1 / 10
+  )
+yield_rec
+```
+Next up is the important package (yes, this is what we named it). When combined with a specific model, we can tune the number of neighbors as well as the proportion of predictors retained (10% above). 
+
+`prep()` will do the appropriate estimation steps: 
+
+```{r}
+#| label: trained-recipe
+trained_rec <- prep(yield_rec)
+```
+
+Which 10% of the predictors were retained? The `tidy()` method can list the scores and their rankings: 
+
+```{r}
+#| label: tidy
+scores <- tidy(trained_rec, number = 2)
+scores |>
+  arrange(desc(.d_overall)) |>
+  select(-starts_with(".d_max_"), -id)
+  
+# What percentage was removed?
+mean(scores$removed * 100)
+```
+
```

Comment:
Some closing sentence about how this extends the preprocessing capabilities in tidymodels would be nice